tests/api_tests/nodes/test_views.py

```diff
@@ -605,6 +610,85 @@ def test_partial_update_private_project_logged_in_non_contributor(self):
         res = self.app.patch_json(self.private_url, {'title': self.new_title}, auth=self.basic_auth_two, expect_errors=True)
         assert_equal(res.status_code, 403)
 
+
+class TestNodeDelete(ApiTestCase):
+
+    def setUp(self):
+        ApiTestCase.setUp(self)
+        self.user = UserFactory.build()
+        self.user.set_password('password')
+        self.user.save()
+        self.basic_auth = (self.user.username, 'password')
+
+        self.project = ProjectFactory(creator=self.user, is_public=False)
+        self.private_url = '/{}nodes/{}/'.format(API_BASE, self.project._id)
+
+        self.user_two = UserFactory.build()
+        self.user_two.set_password('password')
+        self.user_two.save()
+        self.basic_auth_two = (self.user_two.username, 'password')
+
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        self.public_url = '/{}nodes/{}/'.format(API_BASE, self.public_project._id)
+
+        self.fake_url = '/{}nodes/{}/'.format(API_BASE, '12345')
+
+    def test_deletes_public_node_logged_out(self):
+        res = self.app.delete(self.public_url, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
+    def test_deletes_public_node_logged_in(self):
+        res = self.app.delete(self.public_url, auth=self.basic_auth_two, expect_errors=True)
+        assert_equal(res.status_code, 403)
+        assert_equal(self.public_project.is_deleted, False)
+
+        res = self.app.delete(self.public_url, auth=self.basic_auth, expect_errors=True)
+        assert_equal(res.status_code, 202)
+        assert_equal(self.public_project.is_deleted, False)
+        detail = ast.literal_eval(res.json['detail'])
```

Comment:
`u''` shouldn't be a part of any string that's returned from Python - it basically means that `__repr__` is being called on something it shouldn't be.

One place where that happens by accident a lot is when you are expecting a string, get a dict, and explicitly convert it. For some reason, `dict.__str__` just calls `dict.__repr__`. If you ask me, that's a bug :)

Example:

```python
foo = {u'apple': u'pear'}
print(foo)
# {u'apple': u'pear'}

foo_coerced = str(foo)
print(repr(foo_coerced))
# u'{u\'apple\': u\'pear\'}'
```

I'm guessing that's what is happening hear - `str()` is getting called on an dict, when `json.dumps()` should be used instead.